libcrux-ml-dsa/src/key_centric_apis.rs

```diff
@@ -0,0 +1,721 @@
+//! This module includes key-centric and slice-based APIs for ML-DSA.
```

Comment:
This module looks like it's duplicate lots of code and adds way more APIs we need to maintain.
Do you intend to maintain all APIs or drop some?